### ISSUE TITLE
Fastball related bugs and suggestions

### ISSUE THREAD

**Issue by GeckoEidechse** [R2Northstar/NorthstarMods#132]
Based on playing a few Fastball matches in a competitive setting, here's a few things we noticed:

# Bugs:
- [ ] The end of each round seems to have add a death to each player (needs confirmation)
- [ ] Dropped guns are not always deleted between rounds (needs confirmation)\
      On a round on Angel City I rushed B at the start of the round an noticed there was already a gun lying there on the floor, likely from the previous round
- [ ] Players seem to be spawned at the position of the hacker, not the panel (needs confirmation)\
      When hacking B on Glitch I could killed right after hacking, making my camera fall through the floor (related to #55). The majority of my spawned team mates then died due to falling in the river. This implies they were spawned on my current position instead of relative to the panel.
- [ ] Players going invis after respawn on Wargames\
      Tracked in #92

# Suggestions
- [ ] On the scoreboard, instead of deaths, maybe display cumulative players respawned when hacking a panel

# Minor / possibly competitive setting only:

- [ ] Option to reduce BT time and replay\
      This was mentioned on the first game and might be better fitted for 3rd party mod anyway

&nbsp;

(This addresses many points that are all for one specific mode, hence condensing it into one issue instead of multiple separate)

**Comment by abarichello** [comment]
I think the first checkpoint already got addressed here: https://github.com/R2Northstar/NorthstarMods/pull/72

**Comment by Neoministein** [comment]
> * On the scoreboard, instead of deaths, maybe display cumulative players respawned when hacking a panel

Totally agree, would make a lot of sense to see it.

**Comment by GeckoEidechse** [comment]
Added a few more points from today's session.

**Comment by BobTheBob9** [comment]
> Players seem to be spawned at the position of the hacker, not the panel (needs confirmation)
should be fixed as of https://github.com/R2Northstar/NorthstarMods/commit/050e707f4762b976791c0934b88779e038f70ade

**Comment by BobTheBob9** [comment]
> Fastball has no half time yet

should be fixed as of https://github.com/R2Northstar/NorthstarMods/commit/3e98aad76ce9f69eba30770e71518a715991e820

**Comment by GeckoEidechse** [comment]
050e707f4762b976791c0934b88779e038f70ade has been reverted due to causing regressions (#169 #170).

**Comment by Bobbyperson12** [comment]
Hey just wanted to add another bug here, very inconsistently players will respawn with their view in the floor. Other players can see them fine, but these players cannot wallrun or do any movement beyond sprinting (may need additional confirmation). Additionally, you can shoot your gun, but the bullets come out of where your eyes are and not your actual view. Attached is a video of this happening twice. 

https://github.com/R2Northstar/NorthstarMods/assets/23128094/cb441bcf-df4a-4977-b6ce-d08006a9babb

**Comment by GeckoEidechse** [comment]
Thanks for the report <3

What's interesting is that usually this would happen when the player hacking the panel dies at the time players are respawned. However this is clearly not the case in the first clip so really curious what's causing it then ^^"

**Comment by Zanieon** [comment]
> cc @Zanieon for [#132 (comment)](https://github.com/R2Northstar/NorthstarMods/issues/132#issuecomment-1951443648) cause you're an absolute wizard when it comes to this sorta stuff ^^

Judging by the video i can already foresee this problem being related to respawning code not setting up player settings properly and keeping them with spectator settings instead.

I have been doing some refactors of modular core gameplay scripts in my repos and i can take a look into this at some point, granted... if i didn't already fix it with said refactor lmao.

**Comment by Zanieon** [comment]
Took a quick look into Fastball code and i wanna die inside, i pretty much know exactly what causes the bug of respawning as a ground camera

**Comment by Zanieon** [comment]
While working on Extraction gamemode i managed to get the same camera bug on respawn which people called it "rat mode" due to it being on the floor, and due to that i was obliged to track down the source of this problem, which is ultimately the dead players spectating a team mate in third person specifically.

Currently i can't think of any viable way to fix this issue properly because spectators watches teammates from the replay buffer 0.1s in the past, when server respawns them, the signals and commands to clear them from spectator replay and observer mode might be late to be processed, leaving them in said rat mode.

An already existent fix to this is people actually watch the match from the panoramic cameras of the map or a specific player in first person only, third person is guaranteed they will respawn in rat mode.

**Comment by Zanieon** [comment]
Yes it is, i told the people playing with me to fiddle with the spectating modes and the rat mode only specifically happen when watching someone in third person and server respawns said person.

**Comment by Zanieon** [comment]
Yeah i literally tried a quick bandaid fix that works good, which is force players camera into something else for a frame, then respawns them, so far tests with different people on different ping ranges seemed to be consistent in fixing this issue.
